src/Compilers/CSharp/Portable/Binder/Binder_Attributes.cs

```diff
@@ -174,6 +179,90 @@ private BoundAttribute BindAttributeCore(AttributeSyntax node, NamedTypeSymbol a
                 boundNamedArguments, resultKind, attributeType, hasErrors: resultKind != LookupResultKind.Viable);
         }
 
+        private void CheckArgumentNullability(MethodSymbol attributeConstructor, AnalyzedAttributeArguments arguments, DiagnosticBag diagnostics)
+        {
+            // Check constructor arguments
+            for (int i = 0; i < arguments.ConstructorArguments.Arguments.Count; i++)
+            {
+                var argument = arguments.ConstructorArguments.Argument(i);
+                var argName = arguments.ConstructorArguments.Name(i);
+
+                // map the argument back to the correct parameter
+                var parameter = argName == null
+                                ? attributeConstructor.Parameters[i]
+                                : attributeConstructor.Parameters.Single(p => p.Name == argName);
```

Comment:
>Parameters.Single(p => p.Name == argName) [](start = 55, length = 41)

In error cases, might there be zero or more than one parameter with the given name?

Reply:
There is an existing method for matching argument names, `GetMatchingNamedConstructorArgumentIndex`. Should we use that here?

---
In reply to: [262297428](https://github.com/dotnet/roslyn/pull/33816#discussion_r262297428) [](ancestors = 262297428)